provider/boltmem/boltmem_test.go

```diff
@@ -559,3 +559,131 @@ func notifyInfoListEqual(n1, n2 []*types.NotifyInfo) bool {
 	}
 	return true
 }
+
+func TestSilencesQuery(t *testing.T) {
+	var (
+		t0       = time.Now()
+		silences = testNewSilences(t, t0)
+	)
+
+	n := 500
+	insert := make([]*types.Silence, n)
+	for i := 0; i < n; i++ {
+		insert[i] = createNewSilence(t, silences, t0, i)
+	}
+
+	pairs := []queryPair{
+		queryPair{
+			n:      10,
+			offset: 0,
+		},
+		queryPair{
+			n:      10,
+			offset: 2,
+		},
+		queryPair{
+			n:      100,
+			offset: 4,
+		},
+	}
+
+	for _, p := range pairs {
+		res, err := silences.Query(p.n, p.offset)
+		if err != nil {
+			t.Fatalf("Retrieval failed: %s", err)
+		}
+
+		s := res.Silences
+
+		start := p.offset * p.n
+		end := start + p.n
+		if end > n {
+			t.Fatalf("your test data doesn't include the range you're requesting: insert[%d:%d] (max index %d)", start, end, n)
+		}
+		expected := append([]*types.Silence{}, insert[start:end]...)
```

Comment:
can just be `expected := insert[start:end]`. The append would only help if we were potentially modifying the insert slice afterwards. But that's not happening here.